cpp/src/aggregation/aggregation.cu

```diff
@@ -27,8 +27,13 @@ void initialize_with_identity(mutable_table_view& table,
   // TODO: Initialize all the columns in a single kernel instead of invoking one
   // kernel per column
   for (size_type i = 0; i < table.num_columns(); ++i) {
-    auto col = table.column(i);
-    dispatch_type_and_aggregation(col.type(), aggs[i], identity_initializer{}, col, stream);
+    auto col        = table.column(i);
+    auto const type = [&] {
+      if (not is_fixed_point(col.type())) return col.type();
+      return col.type().id() == type_id::DECIMAL32 ? data_type{type_id::INT32}
+                                                   : data_type{type_id::INT64};
+    }();
```

Comment:
Can we just have a runtime `type_id` equivalent of `device_storage_type`?

Reply:
Yea, I had thought about that. Wasn't sure it was worth it as this _currently_ is the only place this is used. Happy to add one though.